src/Jackalope/Transport/TransportInterface.php

```diff
@@ -310,6 +310,10 @@ public function getWeakReferences($path, $name = null);
      */
     public function setNodeTypeManager($nodeTypeManager);
 
+    public function hasNodeType($nodeTypeName);
```

Comment:
also, who will be responsible for caching this? afaik the transport never caches anything, so the OM should probably cache the check result. also, the NodeTypeManager would know if it already loaded the type so it could immediately return true - maybe the method should stay on NodeTypeManager and we give the NTM access to the transport?